processing/src/main/java/io/druid/segment/StringDimensionIndexer.java

```diff
@@ -496,16 +496,19 @@ public void inspectRuntimeShape(RuntimeShapeInspector inspector)
             return BooleanValueMatcher.of(false);
           }
         } else {
-          // Employ precomputed BitSet optimization
+          // Employ caching BitSet optimization
           return makeValueMatcher(Predicates.equalTo(value));
         }
       }
 
       @Override
       public ValueMatcher makeValueMatcher(final Predicate<String> predicate)
       {
-        final BitSet predicateMatchingValueIds = DimensionSelectorUtils.makePredicateMatchingSet(this, predicate);
```

Comment:
Is `DimensionSelectorUtils.makePredicateMatchingSet()` still used or could be deleted?

Reply:
It's still used in one place: ForwardingFilteredDimensionSpec. I would like to rewrite that code to be lazy in a follow up PR, and delete `DimensionSelectorUtils.makePredicateMatchingSet` at that time. I didn't do it in this patch since I wanted to keep it small.